src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/hudsontrigger/GerritTrigger.java

```diff
@@ -548,24 +564,51 @@ protected void schedule(GerritCause cause, GerritTriggeredEvent event) {
      * @param event   the event.
      * @param project the project to build.
      * @deprecated
-     *    moved to {@link EventListener#schedule(GerritTrigger, GerritCause, GerritTriggeredEvent, AbstractProject)}
+     *    moved to {@link EventListener#schedule(GerritTrigger, GerritCause, GerritTriggeredEvent, Job)}
      */
     @Deprecated
-    protected void schedule(GerritCause cause, GerritTriggeredEvent event, AbstractProject project) {
+    protected void schedule(GerritCause cause, GerritTriggeredEvent event, Job project) {
         createListener().schedule(this, cause, event, project);
     }
 
     /**
+     * Schedules a build of a job.
+     * <p>
+     * Added here to facilitate unit testing.
+     *
+     * @param theJob The job.
+     * @param quitePeriod Quite period.
+     * @param cause Build cause.
+     * @param badgeAction build badge action.
+     * @param parameters Build parameters.
+     * @return Scheduled build future.
+     */
+    protected Future schedule(final Job theJob, int quitePeriod, GerritCause cause, BadgeAction badgeAction,
```

Comment:
Hmm, I still don't like it. I put a lot of effort into moving all scheduling logic out of this class not long ago. I need to think about this a bit, either to convince myself that it's OK or find a better way. I think I understand your argument though and I'm all for making the code more testable.